contrib/win32/win32compat/win32_usertoken_utils.c

```diff
@@ -288,7 +282,7 @@ process_custom_lsa_auth(const char* user, const char* pwd, const char* lsa_pkg)
 		goto done;
```

Comment:
Need to fix domain related logic in process_custom_lsa_auth(). It still assumes the older CPN like format.

Reply:
I hadn't looked at this code much before.  It looks like it allows an administrator-provided authentication package by name (via registry key) and then passes in the login information as "username:password" or "username:password:domain".  Based on quick read of the APIs, it looks like this logon info format should be package-dependent.   Was the intent her to target a specific authentication package?  If so, which one since I'd like to test?  If not, is this logon info format documented somewhere as being a standard that custom authentication packages should use?

Reply:
You reverse engineered it right. @bagajjal can you document how this authentication extension can be used.

Reply:
@bagajjal I suppose I really don't need a verbose write-up and/or test it myself, but it would be useful to know what the ideal username/domain format is.  For example, is it SamAccountName:Password:NetBiosName, SamAccountName:Password:DnsDomainName, UpnAccountName:Password:UpnAccountSuffix, or something else entirely?

Reply:
its "username:password:domain".
Here are the steps to test this,
1) download, unzip the [sample lsa provider] (https://github.com/PowerShell/openssh-portable/files/1944211/lsa-provider.zip)
2) from elevated powershell window run ".\install-lsaprov.ps1"
3) Add key "HKLM\SOFTWARE\OpenSSH\LSAAuthenticationPackage" with value "custompwdauth-lsa"
![image](https://user-images.githubusercontent.com/23668037/39208752-b7bff26a-47b8-11e8-8662-a9d26528857d.png)
4) Restart computer.
5) login using ssh - ssh user@ip. Enter dummy password, it should let you login using custom lsa provider configured in step 2.

Reply:
So is this meant to be a theoretical custom provider that somebody could custom-code just for SSH?  Unless I'm missing something, right now the code you sent basically only uses the first token in the "username;password;domain" format. The password and domain parameters aren't used. Moreover, it expects that the username to be in the UserPrincipalName format (aka SecNameFlat for GetAuthDataForUser) which would make the domain field unnecessary (if it were actually being used).  Can you provide some additional context as to what we are actually trying to accomplish with this provider?

Reply:
@NoMoreFood - This is just a sample code used for testing. We have an internal requirement to support custom lsa authentication provider in "username:password:domain" format.

Reply:
@bagajjal Ok, that makes a bit more sense.  The current code actually uses semicolons, not colons and the LSA sample tokenizes on semicolon.  So is it semicolon or colon?  Also, by domain do you mean NetBiosName (DOMAIN) or a fully qualified name (domain.com)?

Reply:
My bad its semicolon, "username;password;domain". Domain is NetBiosName (DOMAIN).

Reply:
@bagajjal My changes are in for this.